env0/resource_aws_credentials.go

```diff
@@ -2,12 +2,36 @@ package env0
 
 import (
 	"context"
+	"regexp"
 
 	"github.com/env0/terraform-provider-env0/client"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
 )
 
+var externalIdRegex *regexp.Regexp = regexp.MustCompile(`^[A-Za-z0-9+=\-.,@:/]{2,1000}$`)
+
+func IsExternalIdValid(d *schema.ResourceData, apiClient client.ApiClientInterface) (bool, error) {
+	isSelfHosted, err := apiClient.IsOrganizationSelfHostedAgent()
+
+	if err != nil {
+		return false, err
+	}
+
+	if isSelfHosted {
```

Comment:
suggestion - to `self-hosted` we have validation to the `secret-reference ` looks like - ${ssm/gcp/azure:<secret-id>}, maybe we can add this validation too

Reply:
I'm unsure we need validation for a field we received from the env0 backend.
Could you explain a little more? (I might be missing some context).